### PR TITLE
[GraphBolt] Update gpu sampling tests of `sample_beighbors`.

### PR DESCRIPTION
**Pull request by yxy235**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by yxy235** [comment]
@dgl-bot

File: tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py [hunk 1]
```diff
@@ -1789,7 +1793,7 @@ def test_sample_neighbors_fanouts(
 
 @unittest.skipIf(
     F._default_context_str == "gpu",
-    reason="Graph is CPU only at present.",
+    reason="Heterogenous sampling on gpu is not supported yet.",
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py:1796]:
```suggestion
    reason="Sampling with replacement not yet supported on GPU.",
```

File: tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py [hunk 2]
```diff
@@ -1848,7 +1852,7 @@ def test_sample_neighbors_replace(
...
 )
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py:1856]:
```suggestion
```

File: tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py [hunk 3]
```diff
@@ -1848,7 +1852,7 @@ def test_sample_neighbors_replace(
...
 @pytest.mark.parametrize("labor", [False, True])
```
Comment (mfbalin) [inline tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py:1857]: Does this test not work for the GPU? What is the error? If you tell me, maybe I can fix it in a quick PR.

Comment (mfbalin) [reply tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py:1857]: Just tested it, it seems to work so we can enable this test for GPU as well.

File: tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py (hunk 2 above)
Comment (mfbalin) [reply tests/python/pytorch/graphbolt/impl/test_fused_csc_sampling_graph.py:1856]: This test seems to work on GPU as well.

**Review by mfbalin — APPROVED** [review]
LGTM. I am working on multi-fanout GPU sampling in #6895 and will enable the hetero tests in that PR.

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**